net: dns-sd: use hostname instead of instance name

toonst opened the pull request:
As described in https://www.rfc-editor.org/rfc/rfc2782, the target field of an SRV record should be the domain name of the target host.

toonst committed:
net: dns-sd: use hostname instead of instance name

As described in https://www.rfc-editor.org/rfc/rfc2782, the target field
of an SRV record should be the host name of the machine providing the
service.

Signed-off-by: Toon Stegen <toon@toostsolutions.be>

jukkar reviewed (commented):
There are compliance issues and CI issues that needs to be fixed.

toonst committed:
tests: net: dns_sd use host name in target field

this fixes the tests to check the hostname is filles in the SRV record's
target field.

Signed-off-by: Toon Stegen <toon@toostsolutions.be>

toonst commented on subsys/net/lib/dns/dns_sd.c:
```diff
@@ -593,9 +594,9 @@ int add_srv_record(const struct dns_sd_rec *inst, uint32_t ttl,
 		/* pointer to .<Instance>.<Service>.<Protocol>.local. */
```
Acutally it is still correct, the full service name is still in the resource record.

toonst commented on subsys/net/lib/dns/dns_sd.c:
```diff
@@ -575,6 +575,7 @@ int add_srv_record(const struct dns_sd_rec *inst, uint32_t ttl,
 	size_t label_size;
```
Yea, the terminology in the RFC is a bit confusing. Updated the commit message to be consistent

jukkar commented on tests/net/lib/dns_sd/src/main.c:
```diff
@@ -382,56 +382,90 @@ ZTEST(dns_sd, test_add_txt_record)
 ZTEST(dns_sd, test_add_srv_record)
```
Typo on test commit msg
s/filles/filled/

cfriedt commented on subsys/net/lib/dns/dns_sd.c:
```diff
@@ -575,6 +575,7 @@ int add_srv_record(const struct dns_sd_rec *inst, uint32_t ttl,
 	size_t label_size;
 	uint16_t inst_offs;
 	uint16_t offset = buf_offset;
+	const char *hostname = net_hostname_get();
```
This function is already pretty testable as a unit. Is there any reason not to add `hostname` as a function parameter?

This change effectively hard-codes the host name, in that it is not easily manipulated by the test harness that tests `add_srv_record()` as a unit.

cfriedt reviewed (changes requested):
Please pass in `hostname` as a parameter to `add_srv_record()`.

The signature of the function will need to be changed in the library as well as in the testsuite to accommodate, but overall doing so will ensure that it can be tested efficiently as a unit by parametrizing inputs.

jukkar commented:
@toonst the requested changes look quite minimal, so just wanted to ask are you able to work on this?
